internal/handler/handler_test.go

```diff
@@ -486,3 +488,146 @@ func TestTransactionManagementService_GetTransactions(t *testing.T) {
 		})
 	}
 }
+func TestTransactionManagementService_DownloadTransaction(t *testing.T) {
```

Comment:
`DownloadTransaction` should have 100% coverage. Looks like some flows may be missing. Confirm.

Reply:
the w.write failure case is not addressed yet

Reply:
done